pandas/tests/tseries/offsets/test_offsets.py

```diff
@@ -574,10 +574,7 @@ def test_add_dt64_ndarray_non_nano(self, offset_types, unit):
 
         result = (dti + off)._with_freq(None)
 
-        exp_unit = unit
-        if isinstance(off, Tick) and off._creso > dti._data._creso:
-            # cast to higher reso like we would with Timedelta scalar
-            exp_unit = Timedelta(off).unit
+        exp_unit = tm.get_finest_unit(dti.unit, Timedelta(off).unit)
```

Comment:
Looks like this change is causing a failure

```bash
FAILED pandas/tests/tseries/offsets/test_offsets.py::TestCommon::test_add_dt64_ndarray_non_nano[BusinessDay0-s] - ValueError: Value must be Timedelta, string, integer, float, timedelta or convertible, not BusinessDay
```

Reply:
woops, reverted